ansible/create_droplets.yml

```diff
@@ -42,97 +45,28 @@
         manager: "{{ manager_created.data.droplet }}"
         workers: "{{ workers_created.results | map(attribute='data.droplet') | list }}"
 
-- name: Configure all Droplets
-  hosts: all
-  tasks:
-    - name: Update apt cache
-      apt:
-        update_cache: yes
-        cache_valid_time: 3600   # Set cache validity time to one hour
+# TEARDOWN (only run when the playbook is run with: --tags teardown)
+    - name: Delete DigitalOcean Manager Droplet
+      digital_ocean_droplet:
+        state: absent
+        name: '{{ manager_name }}'
+        unique_name: true
+        api_token: "{{ DIGITAL_OCEAN_TOKEN }}"
+      when: manager_created is defined
+      tags: teardown,never
 
-    - name: Install docker.io
-      apt:
+    - name: Delete DigitalOcean Worker Droplets
+      digital_ocean_droplet:
+        state: absent
         name: "{{ item }}"
-        state: present
-      loop:
-        - docker.io
-
-- name: Configure manager Droplet
-  hosts: manager
-  tasks:
-    - name: Install docker compose
-      apt:
-        name: docker-compose-v2
-        state: present
-
-    - name: Allow necessary ports in UFW
-      become: yes
-      ufw:
-        rule: allow
-        port: "{{ item.split('/')[0] }}"   # Extract port number from the item
-        proto: "{{ item.split('/')[1] }}"  # Extract protocol from the item
-        state: enabled
-      with_items:
-        - "22/tcp" # SSH
-        - "2376/tcp"
-        - "2377/tcp"
-        - "7946/udp"
-        - "7946/tcp"
-        - "4789/udp"
-
-    - name: Sync remote_files folder (pushes files to the remote server)
-      ansible.posix.synchronize:
-        src: ../remote_files # path relative this the playbook
-        dest: /
-
-    # MIDLERTIDIG START -->
-    - name: Simulate vagrant_prom_data (fix migration)
-      file:
-        path: /remote_files/vagrant_prom_data
-        state: directory
-
-    - name: Simulate vagrant_grafana_storage (fix migration)
-      file:
-        path: /remote_files/vagrant_grafana_storage
-        state: directory
-    # <-- MIDLERTIDIG SLUT
-    
-    - name: Restart Docker service
-      systemd:
-        name: docker
-        state: restarted
-
-    - name: Initialize Docker Swarm
-      shell: docker swarm init --advertise-addr {{ ansible_host }}
-      register: swarm_init_output
-      failed_when: swarm_init_output.failed and swarm_init_output.changed
-
-    - name: Extract Swarm Join Token
-      shell: docker swarm join-token worker -q
-      register: swarm_join_token
-
-    - set_fact:
-        SWARM_JOIN_TOKEN: "{{ swarm_join_token.stdout }}"
-        SWARM_MANAGER_IP: "{{ ansible_host }}"
-
+        unique_name: true
+        api_token: "{{ DIGITAL_OCEAN_TOKEN }}"
+      loop: "{{ workers_created.results | map(attribute='data.droplet.name') | list }}"
+      when: workers_created is defined
+      tags: teardown,never
```

Comment:
What does the `never` tag do here?

Reply:
that it is only run when specifying "--tags teardown" when running the playbook

Reply:
Ah yes, so you guessed correctly